src/com/palmergames/bukkit/towny/Towny.java

```diff
@@ -483,183 +477,100 @@ public void onDisable() {
-			papiExpansion.register();
-            addons.add(String.format("%s v%s", "PlaceholderAPI", test.getDescription().getVersion()));
-		}
+		if (Bukkit.getPluginManager().getPlugin("PlaceholderAPI") != null)
+			new TownyPlaceholderExpansion(this).register();
 		
-		test = getServer().getPluginManager().getPlugin("LuckPerms");
-		if (test != null && TownySettings.isContextsEnabled()) {
-			this.luckPermsContexts = new LuckPermsContexts(this);
-			luckPermsContexts.registerContexts();
-			addons.add(String.format("%s v%s", "LuckPerms", test.getDescription().getVersion()));
-		}
-
-		//Add our chat handler to TheNewChat via the API.
-		if(Bukkit.getPluginManager().isPluginEnabled("TheNewChat")) {
+		if (Bukkit.getPluginManager().isPluginEnabled("TheNewChat"))
 			TNCRegister.initialize();
-		}
-
-		/*
-		 * Test for Citizens2 so we can avoid removing their NPC's
-		 */
-		setCitizens2(getServer().getPluginManager().isPluginEnabled("Citizens"));
-
-		/*
-		 * Output discovered plugins and warnings.
-		 */
-		plugin.getLogger().info("Plugins found: ");
-		
-		for (String string : permissions.split("\n"))
-			plugin.getLogger().info(string);
 		
-		if (!economy.isEmpty())
-			plugin.getLogger().info(economy);
-		if (!addons.isEmpty())
-			plugin.getLogger().info("  Add-ons: " + StringMgmt.wrap(String.join(", ", addons), 52, System.lineSeparator() + "                           "));
-		if (!ecowarn.isEmpty())
-			plugin.getLogger().info(StringMgmt.wrap(ecowarn, 55, System.lineSeparator() + "                           "));
-
-		//Add our chat handler to TheNewChat via the API.
-		if(Bukkit.getPluginManager().isPluginEnabled("TheNewChat")) {
-			TNCRegister.initialize();
-		}
-
-		//Legacy check to see if questioner.jar is still present.
-		test = getServer().getPluginManager().getPlugin("Questioner");
-		if (test != null) {
-			String questioner = "  Warning: Questioner.jar present on server, Towny no longer requires Questioner for invites/confirmations. You may safely remove Questioner.jar from your plugins folder.";
-			plugin.getLogger().warning(StringMgmt.wrap(questioner, 55, System.lineSeparator() + "                           "));
-		}
+		if (Bukkit.getPluginManager().isPluginEnabled("Citizens2"))
+			setCitizens2(true);
 
-		//Add warning about PowerRanks.
-		test = getServer().getPluginManager().getPlugin("PowerRanks");
-		if (test != null) {
-			String powerranks = "  Warning: PowerRanks is incompatible with Towny. PowerRanks will override Towny's ability to give permissions via the townyperms.yml file."
-							+ " You can expect issues with Towny permissions (and other permission providers,) while PowerRanks is installed.";
-			plugin.getLogger().warning(StringMgmt.wrap(powerranks, 55, System.lineSeparator() + "                           "));
+		if (!discovered.isEmpty()) {
+			plugin.getLogger().info(Translation.of("msg_compat_discovered", discovered.size(), String.join(", ", discovered)));
```

Comment:
This should be '  Add-ons" ' and then use teh StringMgmt.wrap() method like the current system uses.

Reply:
I believe you're still missing version numbers as well.